src/components/CreateItem.jsx

```diff
@@ -1,18 +1,89 @@
-import React from 'react';
-import firebase from 'firebase/app';
+import React, { useState } from 'react';
+import db from '../lib/firebase';
+import './createItem.css';
 
 function CreateItem() {
-  const handleClick = async () => {
-    await firebase.firestore().collection('items').add({
-      name: 'A test product',
+  const itemObj = {
+    itemName: '',
+    frequency: 7,
+  };
+  const [item, setItem] = useState(itemObj);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const token = localStorage.getItem('token');
+
+    db.collection(token).add({
+      name: item.itemName,
+      frequency: item.frequency,
+      lastPurchasedDate: null,
       date: Date().toLocaleString(),
     });
+    setItem(itemObj);
+  };
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setItem({
+      ...item,
+      [name]: value,
+    });
   };
 
   return (
     <div>
       <h1>AddView</h1>
-      <button onClick={() => handleClick()}>Create item</button>
+
+      <form onSubmit={handleSubmit}>
+        <label htmlFor="item-name">Item name:</label>
+        <input
+          type="text"
+          id="item-name"
+          value={item.itemName}
+          name="itemName"
+          onChange={handleChange}
+        />
+        <fieldset className="radio-buttons">
+          <legend>How soon will you buy this again?</legend>
+          <div>
+            <input
+              type="radio"
+              id="soon"
+              value={7}
+              name="frequency"
+              onChange={handleChange}
+              checked={item.frequency === '7'}
```

Comment:
Hmm, I did not remember about `checked` attributes existence, and just googled 😅 It is a great addition for input fields 👍 

I'm guessing you are using it to pre-select (check) the input field? If yes, I think you can keep just `checked`, without `{item.frequency === '7'}`. I also think that if the intention is to have only one input pre-selected, you can remove the rest of `checked` attributes from other input fields 🙌

Reply:
Hi @Lukabaramishvili , we did try this approach initially, however React kept re-setting the checked input to the one with the "checked" attribute set, so the visually checked item and the React state were out of line. We chose this approach so that the react state and form input would always match. If you know a way around the checked option with React though, please let us know 😃

Reply:
ah, I see. You do not want to have any items to be checked by default? If that is the intention, it totally works 👍

Reply:
Hi Luka, I see what you mean now! I've managed to fix it by changing the default itemObj to have the frequency set as a string, and now it shows the first one checked on load :)
```
const itemObj = {
    itemName: '',
    frequency: '7',
  };
```
Thanks!